The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9883017552/job/27296971587) [(plain)](https://github.com/rust-lang/rust/commit/3f385945ab941dd1e1668471b4f97558213365bb/checks/27296971587/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.423   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.443 Collecting boolean-py==4.0
#13 0.451   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.468 Collecting chardet==5.1.0
---
#13 3.577 Building wheels for collected packages: reuse
#13 3.578   Building wheel for reuse (pyproject.toml): started
#13 3.898   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.899   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.899   Stored in directory: /tmp/pip-ephem-wheel-cache-gwfzdblw/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.902 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.924   Attempting uninstall: setuptools
#13 3.924     Found existing installation: setuptools 59.6.0
#13 3.926     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.926     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.926     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.595 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.595 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.094 Collecting virtualenv
#13 5.148   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.370      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 25.9 MB/s eta 0:00:00
#13 5.411 Collecting distlib<1,>=0.3.7
#13 5.418   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.429      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 48.3 MB/s eta 0:00:00
#13 5.461 Collecting platformdirs<5,>=3.9.1
#13 5.468   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.504 Collecting filelock<4,>=3.12.2
#13 5.511   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.598 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.769 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 5.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      196544 kB
DirectMap2M:     7143424 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/0c81f94b9a6207fb1fc080caa83584dea2d71fc6/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-0c81f94b9a6207fb1fc080caa83584dea2d71fc6-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 30.11s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/core/src/iter/sources/repeat.rs at line 7:
 ///
 /// Infinite iterators like `repeat()` are often used with adapters like
 /// [`Iterator::take()`], in order to make them finite.
-/// 
+///
 /// > Please prefer [`str::repeat()`] over this function if you just want
 /// > to repeat a char/string `n`th times.
 ///
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/num/dec2flt/decimal.rs" "/checkout/library/std/src/os/wasi/ffi.rs" "/checkout/library/core/src/num/dec2flt/table.rs" "/checkout/library/core/src/num/dec2flt/number.rs" "/checkout/library/core/src/num/dec2flt/mod.rs" "/checkout/library/core/src/num/dec2flt/common.rs" "/checkout/library/core/src/num/dec2flt/parse.rs" "/checkout/library/core/src/num/dec2flt/fpu.rs" "/checkout/library/core/src/num/dec2flt/float.rs" "/checkout/library/core/src/num/dec2flt/slow.rs" "/checkout/library/core/src/num/dec2flt/lemire.rs" "/checkout/library/core/src/num/overflow_panic.rs" "/checkout/library/core/src/primitive.rs" "/checkout/library/core/src/panicking.rs" "/checkout/library/core/src/range.rs" "/checkout/library/core/src/hint.rs" "/checkout/library/core/src/time.rs" "/checkout/library/core/src/future/ready.rs" "/checkout/library/core/src/future/future.rs" "/checkout/library/core/src/future/poll_fn.rs" "/checkout/library/core/src/future/mod.rs" "/checkout/library/core/src/future/join.rs" "/checkout/library/core/src/future/into_future.rs" "/checkout/library/core/src/future/async_drop.rs" "/checkout/library/core/src/future/pending.rs" "/checkout/library/core/src/panic/location.rs" "/checkout/library/core/src/panic/panic_info.rs" "/checkout/library/core/src/panic/unwind_safe.rs" "/checkout/library/core/src/primitive_docs.rs" "/checkout/library/core/src/cell/once.rs" "/checkout/library/core/src/cell/lazy.rs" "/checkout/library/core/src/task/ready.rs" "/checkout/library/core/src/task/wake.rs" "/checkout/library/core/src/task/mod.rs" "/checkout/library/core/src/task/poll.rs" "/checkout/library/core/src/macros/mod.rs" "/checkout/library/core/src/iter/sources/from_fn.rs" "/checkout/library/core/src/iter/sources/from_coroutine.rs" "/checkout/library/core/src/iter/sources/repeat_n.rs" "/checkout/library/core/src/iter/sources/repeat_with.rs" "/checkout/library/core/src/iter/sources/once_with.rs" "/checkout/library/core/src/iter/sources/successors.rs" "/checkout/library/core/src/iter/sources/repeat.rs" "/checkout/library/core/src/iter/sources/once.rs" "/checkout/library/core/src/iter/sources/empty.rs" "/checkout/library/core/src/iter/mod.rs" "/checkout/library/core/src/iter/adapters/scan.rs" "/checkout/library/core/src/iter/adapters/cloned.rs" "/checkout/library/core/src/iter/adapters/filter.rs" "/checkout/library/core/src/iter/adapters/chain.rs" "/checkout/library/core/src/iter/adapters/peekable.rs" "/checkout/library/core/src/iter/adapters/map_windows.rs" "/checkout/library/core/src/iter/adapters/take.rs" "/checkout/library/core/src/iter/adapters/take_while.rs" "/checkout/library/core/src/iter/adapters/inspect.rs" "/checkout/library/core/src/iter/adapters/mod.rs" "/checkout/library/core/src/iter/adapters/by_ref_sized.rs" "/checkout/library/core/src/iter/adapters/map.rs" "/checkout/library/core/src/iter/adapters/zip.rs" "/checkout/library/core/src/iter/adapters/copied.rs" "/checkout/library/core/src/iter/adapters/step_by.rs" "/checkout/library/core/src/iter/adapters/flatten.rs" "/checkout/library/core/src/iter/adapters/intersperse.rs" "/checkout/library/std/src/os/wasi/io/fd.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Jul 10 23:50:19 UTC 2024
  network time: Wed, 10 Jul 2024 23:50:19 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/iter/sources/repeat.rs
+++ b/library/core/src/iter/sources/repeat.rs
@@ -8,11 +8,15 @@ use crate::num::NonZero;
 /// Infinite iterators like `repeat()` are often used with adapters like
 /// [`Iterator::take()`], in order to make them finite.
 ///
+/// Use [`str::repeat()`] instead of this function if you just want to repeat
+/// a char/string `n`th times.
+///
 /// If the element type of the iterator you need does not implement `Clone`,
 /// or if you do not want to keep the repeated element in memory, you can
 /// instead use the [`repeat_with()`] function.
 ///
 /// [`repeat_with()`]: crate::iter::repeat_with
+/// [`str::repeat()`]: ../../std/primitive.str.html#method.repeat
 ///
 /// # Examples
 ///

